# 产生目录

目录中英文映射：

In [38]:
folders = {
  '01-ruby-overview' => '01. **Ruby 快速概览**',
  '02-ruby-basics' => '02. **Ruby 基础**',
  '03-ruby-syntax' => '03. **Ruby 语法进阶**',
  '04-ruby-oop' => '04. **Ruby 面向对象编程**',
  '05-ruby-advanced' => '05. **Ruby 进阶**',
  '06-ruby-on-rails' => '06. **Ruby On Rails**'
}

{"01-ruby-overview"=>"01. **Ruby 快速概览**", "02-ruby-basics"=>"02. **Ruby 基础**", "03-ruby-syntax"=>"03. **Ruby 语法进阶**", "04-ruby-oop"=>"04. **Ruby 面向对象编程**", "05-ruby-advanced"=>"05. **Ruby 进阶**", "06-ruby-on-rails"=>"06. **Ruby On Rails**"}

产生目录文件：

In [41]:
require 'json'
def get_title(ipynb_file, default)
  obj = JSON.parse(File.read(ipynb_file))
  if obj and obj['cells'] and obj['cells'][0] and obj['cells'][0]['source'] and obj['cells'][0]['source'][0]
    title = obj['cells'][0]['source'][0].chomp.sub(/\#\s*/, '')
    index = default.match(/^[\d.]+/)
    "#{index}.**#{title}**"
  else
    default
  end
end

:get_title

In [43]:
open("index.md", 'w') do|io|
  folders.each do|folder, folder_name|
    io << "- [#{folder_name}](#{folder})\n"
    files = Dir.glob("#{folder}/*.ipynb")
    files.sort.each do|file|
      io << "\t- [#{get_title(file, File.basename(file))}](#{file})\n"
    end
  end
end

puts File.read("index.md")

- [01. **Ruby 快速概览**](01-ruby-overview)
	- [01.01.**Ruby 简介**](01-ruby-overview/01.01-ruby-overview.ipynb)
	- [01.02.**20分钟体验 Ruby**](01-ruby-overview/01.02-ruby-in-20-minutes.ipynb)
	- [01.03.**从其它语言到 Ruby**](01-ruby-overview/01.03-ruby-from-other-languages.ipynb)
- [02. **Ruby 基础**](02-ruby-basics)
	- [02.01.**Ruby 数据类型**](02-ruby-basics/02.01-types.ipynb)
	- [02.02.**数字**](02-ruby-basics/02.02-numbers.ipynb)
	- [02.03.**字符串**](02-ruby-basics/02.03-strings.ipynb)
	- [02.04.**符号**](02-ruby-basics/02.04-symbols.ipynb)
	- [02.05.**正则表达式**](02-ruby-basics/02.05-regular-expression.ipynb)
	- [02.06.**列表**](02-ruby-basics/02.06-arrays.ipynb)
	- [02.07.**Hash**](02-ruby-basics/02.07-hashes.ipynb)
	- [02.08.**范围 (Range)**](02-ruby-basics/02.08-ranges.ipynb)
- [03. **Ruby 语法进阶**](03-ruby-syntax)
	- [03.01.**Ruby 赋值机制**](03-ruby-syntax/03.01-assignment-statements.ipynb)
	- [03.02.**判断语句**](03-ruby-syntax/03.02-conditional-statements.ipynb)
	- [03.03.**循环**](03-ruby-syntax/03.03-loop-statements.

- [01. **Ruby 快速概览**](01-ruby-overview)
	- [01.01.**Ruby 简介**](01-ruby-overview/01.01-ruby-overview.ipynb)
	- [01.02.**20分钟体验 Ruby**](01-ruby-overview/01.02-ruby-in-20-minutes.ipynb)
	- [01.03.**从其它语言到 Ruby**](01-ruby-overview/01.03-ruby-from-other-languages.ipynb)
- [02. **Ruby 基础**](02-ruby-basics)
	- [02.01.**Ruby 数据类型**](02-ruby-basics/02.01-types.ipynb)
	- [02.02.**数字**](02-ruby-basics/02.02-numbers.ipynb)
	- [02.03.**字符串**](02-ruby-basics/02.03-strings.ipynb)
	- [02.04.**符号**](02-ruby-basics/02.04-symbols.ipynb)
	- [02.05.**正则表达式**](02-ruby-basics/02.05-regular-expression.ipynb)
	- [02.06.**列表**](02-ruby-basics/02.06-arrays.ipynb)
	- [02.07.**Hash**](02-ruby-basics/02.07-hashes.ipynb)
	- [02.08.**范围 (Range)**](02-ruby-basics/02.08-ranges.ipynb)
- [03. **Ruby 语法进阶**](03-ruby-syntax)
	- [03.01.**Ruby 赋值机制**](03-ruby-syntax/03.01-assignment-statements.ipynb)
	- [03.02.**判断语句**](03-ruby-syntax/03.02-conditional-statements.ipynb)
	- [03.03.**循环**](03-ruby-syntax/03.03-loop-statements.ipynb)
	- [03.04.**迭代器**](03-ruby-syntax/03.04-iterators.ipynb)
	- [03.05.**函数**](03-ruby-syntax/03.05-functions.ipynb)
	- [03.09.**异常**](03-ruby-syntax/03.09-exceptions.ipynb)
	- [03.10.**Ruby元编程技巧**](03-ruby-syntax/03.10-ruby-meta-programming.ipynb)
- [04. **Ruby 面向对象编程**](04-ruby-oop)
- [05. **Ruby 进阶**](05-ruby-advanced)
- [06. **Ruby On Rails**](06-ruby-on-rails)